packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -812,6 +864,56 @@ export class Project extends ProjectBase {
       : this._secondaryArtifacts.map((secondaryArtifact) => secondaryArtifact.toArtifactsJSON());
   }
 
+  /**
+   * If configured, set up the VPC-related properties
+   *
+   * Returns the VpcConfig that should be added to the
+   * codebuild creation properties.
+   */
+  private configureVpc(props: ProjectProps): CfnProject.VpcConfigProperty | undefined {
+    if ((props.securityGroups || props.allowAllOutbound !== undefined) && !props.vpc) {
+      throw new Error(`Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC`);
+    }
+
+    if (!props.vpc) { return undefined; }
+
+    if ((props.securityGroups && props.securityGroups.length > 0)  && props.allowAllOutbound !== undefined) {
+      throw new Error(`Configure 'allowAllOutbound' directly on the supplied SecurityGroup.`);
+    }
+
+    if (props.securityGroups && props.securityGroups.length > 0) {
+      this._securityGroups = [...this._securityGroups, ...props.securityGroups];
+    } else {
+      const securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
+        vpc: props.vpc,
+        description: 'Automatic generated security group for CodeBuild ' + this.node.uniqueId,
+        allowAllOutbound: props.allowAllOutbound
+      });
+      this._securityGroups.push(securityGroup);
+    }
+    const vpcPlacement: ec2.VpcPlacementStrategy = props.vpcPlacement ? props.vpcPlacement : {
+      subnetsToUse: ec2.SubnetType.Private
+    };
+    const policyStatement = new iam.PolicyStatement(iam.PolicyStatementEffect.Allow)
+      .addAllResources()
+      .addActions(
+      'ec2:CreateNetworkInterface',
+      'ec2:DescribeNetworkInterfaces',
+      'ec2:DeleteNetworkInterface',
+      'ec2:DescribeSubnets',
+      'ec2:DescribeSecurityGroups',
+      'ec2:DescribeDhcpOptions',
+      'ec2:DescribeVpcs',
+      'ec2:CreateNetworkInterfacePermission'
```

Comment:
Indentation is off here, as well as above and the one line below.

Reply:
It would also be amazing if you included the condition on the `CreateNetworkInterfacePermission` mentioned in the docs: https://docs.aws.amazon.com/codebuild/latest/userguide/auth-and-access-control-iam-identity-based-access-control.html#customer-managed-policies-example-create-vpc-network-interface

That's just bonus points though, I won't block the CR on it :)

Reply:
Fixed and implemented ;)